lib/test.js

```diff
@@ -28,6 +29,15 @@ function Test(title, fn) {
   Runnable.call(this, title, fn);
   this.pending = !fn;
   this.type = 'test';
+
+  /**
+   * @summary State of Test after execution
+   * @desc One of `Test.constants.STATE_FAILED` or `Test.constants.STATE_PASSED`.
+   * @see Test.constants
+   * @type {string}
+   * @public
+   */
+  this.state = void 0;
```

Comment:
Why not `undefined` or `null`, especially as its supposedly a string?

Reply:
habit.  `undefined` can be redefined in older browsers.  we don't support any of those anymore (IE11 disallows this), so `undefined` is safe to use.

- [ ] use `undefined`

`null` needs to die.

Reply:
hmm.   a problem with a truthy default value for `Test#state` is that some 3rd party reporter might have code that does this:

```js
runner.on('test end', test => {
  if (test.state && test.state !== 'failed') {
    // something
  }
});
```

adding another state, e.g. `test.state = 'init'` would break this assumption.

the reason for setting `test.state = something` initially is really just... not a good one.  I'm unsure how to document an instance variable without an actual assignment statement.  that's the reason.  maybe best to punt on it until/if that becomes clear.

Reply:
*punting*